src/index.js

```diff
@@ -46,25 +46,32 @@ function runSync (command, options) {
 }
 
 function runAsync (command, options) {
-  // Prepare options for exec command (don't need async and stdio as it doesn't handle them)
-  const execOptions = Object.assign({}, options)
-  delete execOptions.async
-  delete execOptions.stdio
-
   return new Promise((resolve, reject) => {
-    const asyncProcess = childProcess.exec(command, execOptions, (error, stdout) => {
-      if (error) {
-        reject(error)
-      } else {
-        resolve(stdout.toString())
+    const spawnOptions = Object.assign({shell: true}, options)
+    const timeout = spawnOptions.timeout
+    delete spawnOptions.async
+
+    return new Promise((resolve, reject) => {
+      const asyncProcess = childProcess.spawn(command, spawnOptions)
+      asyncProcess.on('error', (error) => {
+        reject(new Error(`Failed to start command: ${command}; ${error}`))
+      })
+
+      asyncProcess.on('close', (exitCode) => {
+        if (exitCode === 0) {
+          resolve(exitCode)
```

Comment:
Yeah we actually can't return final data from `spawn`, because it's streaming data not gather them in a buffer like `exec`. We would need to simulate that behaviour. Idea would be to gather data to a "leaky" buffer: 

```javascript
const buffer = []
const MAX = 100
asyncProcess.stdout.on('data', (data) => {
  if(buffer.length > MAX) {
    buffer.shift() // prevent buffer exceeding
  }
  buffer.push(data)
})
```

after that we could resolve in `exitCode === 0` with for example `resolve(buffer.join('\n'))`. Example is not perfect but it gives an idea. Buffer will be leaky, so it will store let say only 100 last items. Buffer exceeding error will not be present, `runAsync` will resolve but the trade off is that some data will be lost if buffer gets out of max storage.